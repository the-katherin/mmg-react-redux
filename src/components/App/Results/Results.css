.results-window {
  display: grid;
  justify-content: center;
  justify-items: center;
  grid-row-gap: 20px;
  padding: 30px 50px;
  border-radius: 10px;
  border: 1px solid white;
  background: rgba(0, 120, 201, 0.5);
  
}

.results__list {
    display: grid;
    grid-row-gap: 10px;
    justify-content: center;
    justify-items: center;

}








