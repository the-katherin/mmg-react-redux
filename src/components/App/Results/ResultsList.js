import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import { selectTopResults } from '../../../selectors'


export class ResultsList extends Component {

    render() {

        const topResults = this.props.topResults;


        return (
            <ol className='results__list'>
                {
                    topResults.map(
                        (item, index) =>
                            <li key={index}>
                                {item.username}: {item.score}
                            </li>)
                }
            </ol>
        );
    }
};


ResultsList.propTypes = {
    topResults: PropTypes.oneOfType([PropTypes.array, PropTypes.string]).isRequired,
};


export default connect(
    selectTopResults
)(ResultsList);