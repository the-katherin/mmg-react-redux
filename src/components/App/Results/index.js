import './Results.css';

export { default } from './Results';
export { Results } from './Results';
export { ResultsList } from './ResultsList';