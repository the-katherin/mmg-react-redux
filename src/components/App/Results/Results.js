import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {
    getUserData,
    getRequestStatus,
    getUserScore,
    getIsFinished
} from "../../../selectors";
import ResultsList from './ResultsList';
import { fetchAllScores, postScore, clearGameMemory } from '../../../actions'

export class Results extends Component {


    async componentDidMount() {
        if (this.props.isFinished) {
            await this.props.postScore(
                this.props.userData.name + this.props.userData.lastName,
                this.props.userData.email,
                this.props.userScore
            );

        };

        this.props.fetchAllScores();

        this.props.clearGameMemory();

    }

    render() {

        const requestStatus = this.props.requestStatus;

        let response;

        switch (requestStatus) {
            case 'isFetching':
                response = <div> Loading... </div>;
                break;
            case 'success':
                response = <ResultsList />;
                break;
            case 'error':
                response = <div> Sorry, an error on server occured </div>;
                break;
            default:
                response = '';
        }


        return (
            <div className="results-window">

                {this.props.userScore > 0 && <h2> You did it! Welldone! Your score is {this.props.userScore}</h2>}

                <h3>TOP-10 results </h3>
                {response}

                <Link to={"/game"} ><button>Restart</button></Link>
            </div>
        )
    }
};

Results.propTypes = {
    userData: PropTypes.shape({
        name: PropTypes.string,
        lastName: PropTypes.string,
        email: PropTypes.string
    }).isRequired,
    userScore: PropTypes.number,
    requestStatus: PropTypes.string.isRequired,
    postScore: PropTypes.func.isRequired,
    fetchAllScores: PropTypes.func.isRequired,
};

export default connect(
    state => ({
        userData: getUserData(state),
        userScore: getUserScore(state),
        requestStatus: getRequestStatus(state),
        isFinished: getIsFinished(state)

    }),
    {
        clearGameMemory,
        postScore,
        fetchAllScores,

    }
)(Results);


