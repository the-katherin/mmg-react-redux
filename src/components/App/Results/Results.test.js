import React from "react";
import { Results } from "./Results";

let wrapper;

const clearResultsMemory = jest.fn();
const postScore = jest.fn();
const fetchAllScores = jest.fn();

describe("Results", () => {
    beforeEach(() => {
        wrapper = shallow(<Results
            clearResultsMemory={clearResultsMemory}
            postScore={postScore}
            fetchAllScores={fetchAllScores}
            userData={{ name: '', lastName: '', email: '' }}
            userScore={5}
            requestStatus={'isFetching'}


        />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });


});
