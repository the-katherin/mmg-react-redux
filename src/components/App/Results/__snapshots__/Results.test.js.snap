// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results match snapshot 1`] = `
<div
  className="results-window"
>
  <h2>
     You did it! Welldone! Your score is 
    5
  </h2>
  <h3>
    TOP-10 results 
  </h3>
  <div>
     Loading... 
  </div>
  <Link
    replace={false}
    to="/game"
  >
    <button>
      Restart
    </button>
  </Link>
</div>
`;
